packages/framework/global/src/utils/math.ts

```diff
@@ -536,3 +537,11 @@ export function getCenterAreaBounds(bounds: IBound, ratio: number) {
     rotate,
   };
 }
+
+export function dotProduct(vectorA: IPoint, vectorB: IPoint) {
```

Comment:
Please use `Vec.dpr` and `Vec.len` instead.

Reply:
thx, replaced